public/js/components/Editor.js

```diff
@@ -174,25 +176,62 @@ const Editor = React.createClass({
   componentWillReceiveProps(nextProps) {
     // This lifecycle method is responsible for updating the editor
     // text.
-    const sourceText = nextProps.sourceText;
+    const { sourceText, selectedLocation } = nextProps;
 
     if (!sourceText) {
-      this.setText("");
-      this.editor.setMode({ name: "text" });
-    } else if (!isTextForSource(sourceText)) {
-      // There are only 2 possible states: errored or loading. Do
-      // nothing except put a message in the editor.
-      this.setText(sourceText.get("error") || "Loading...");
-      this.editor.setMode({ name: "text" });
-    } else if (this.props.sourceText !== sourceText) {
-      // Only update it if the `sourceText` object has actually changed.
-      // It is immutable so it will always change when updated.
-      this.setText(sourceText.get("text"));
-      this.setMode(sourceText);
```

Comment:
Yeah, it shouldn't take much more. I'd like to try to maintain the same code flow as before and keep the docs local, and I'll make a quick commit to show you what I mean. I'm not sure, but it might be a little cleaner.